arangod/Pregel/Worker.cpp

```diff
@@ -169,31 +169,14 @@ void Worker<V, E, M>::setupWorker() {
     _feature.metrics()->pregelWorkersLoadingNumber->fetch_sub(1);
   };
 
-  std::function<void()> statusUpdateCallback = [self = shared_from_this(),
-                                                this] {
-    VPackBuilder statusUpdateMsg;
-    {
-      auto ob = VPackObjectBuilder(&statusUpdateMsg);
-      statusUpdateMsg.add(Utils::senderKey,
-                          VPackValue(ServerState::instance()->getId()));
-      statusUpdateMsg.add(Utils::executionNumberKey,
-                          VPackValue(_config.executionNumber()));
-      statusUpdateMsg.add(VPackValue(Utils::payloadKey));
-      auto update = observeStatus();
-      serialize(statusUpdateMsg, update);
-    }
-
-    _callConductor(Utils::statusUpdatePath, statusUpdateMsg);
-  };
-
   // initialization of the graphstore might take an undefined amount
   // of time. Therefore this is performed asynchronously
   TRI_ASSERT(SchedulerFeature::SCHEDULER != nullptr);
   _feature.metrics()->pregelWorkersLoadingNumber->fetch_add(1);
   Scheduler* scheduler = SchedulerFeature::SCHEDULER;
   scheduler->queue(RequestLane::INTERNAL_LOW,
                    [this, self = shared_from_this(),
-                    statusUpdateCallback = std::move(statusUpdateCallback),
+                    statusUpdateCallback = std::move(_statusCallback()),
```

Comment:
just realized that it should probably be _statusCallback and not _statusCallback()

Reply:
it actually needs to be `_statusCallback()`, because `_statusCallback` returns the function to be called as status callback; I renamed the function to `_makeStatusCallback` to make this less obscure; we should though have a better design for these updates for later versions of Pregel.